AWSIoTPythonSDK/core/protocol/paho/client.py

```diff
@@ -780,7 +782,9 @@ def reconnect(self):
         self._messages_reconnect_reset()
 
         try:
-            if (sys.version_info[0] == 2 and sys.version_info[1] < 7) or (sys.version_info[0] == 3 and sys.version_info[1] < 2):
+            if self._socket_factory:
+                sock = self._socket_factory()
```

Comment:
I feel like the socket factory lambda should take the same arguments as connect: host, port, timeout, source_address.  You'd pass the appropriate values from the client object.

It can work either way, but if the factory takes no arguments then you have to embed your connect() arguments into it which feels overly coupled.

Reply:
I might misunderstand what you mean, but the socket_factory is not injecting arguments into the socket.create_connection, it's replacing that method entirely. Since we are using a completely different method for creating a socket connection, wouldn't it make sense that all arguments associated with the new factory function be sent as together? If there was another socket library with different arguments for example, wouldn't it block that use case?

Reply:
You may be right.  I'm going to get another pair of eyes on this and if it's all ok, we'll be merging via an internal repository.  The changes will eventually show up when there's a new version pushed here.

Reply:
Can I get a rough ETA? My company is very interested in having this ability merged as soon as possible.